libgit/browser.go

```diff
@@ -228,9 +238,20 @@ func (b *Browser) Join(elem ...string) string {
 
 // ReadDir implements the billy.Filesystem interface for Browser.
 func (b *Browser) ReadDir(p string) (fis []os.FileInfo, err error) {
+	if p == "" {
+		p = "."
+	}
+
+	cachePath := path.Join(b.root, p)
```

Comment:
Also `path.Clean` this path?

Reply:
`path.Join` already does a `Clean` before returning: https://golang.org/src/path/path.go?s=4034:4066#L145 so no need here